modules/seqexec/web/server/src/main/scala/seqexec/web/server/http4s/WebServerLauncher.scala

```diff
@@ -190,33 +229,38 @@ object WebServerLauncher extends StreamApp[IO] with LogInitialization with Seqex
         met        <- SeqexecMetrics.build[IO](site, collector)
       } yield SeqexecEngine(httpClient, seqc, met)
 
-    def webServerIO(in: Queue[IO, executeEngine.EventType], out: Topic[IO, SeqexecEvent], et: SeqexecEngine, pe: PrometheusExportService[IO]): IO[Stream[IO, ExitCode]] =
-      // Launch web server
+    def webServerIO(
+      in:  Queue[IO, executeEngine.EventType],
+      out: Topic[IO, SeqexecEvent],
+      et:  SeqexecEngine,
+      cr:  CollectorRegistry,
+      bec: ExecutionContext
+    ): Resource[IO, Unit] =
       for {
-        wc <- serverConf
-        ac <- authConf.run(wc)
-        as <- authService.run(ac)
-        _  <- logStart.run(wc)
-        _  <- logToClients(out)
-      } yield Stream(redirectWebServer.run(wc), webServer(as, in, out, et, pe).run(wc)).join(2)
-
-    // I have taken this from the examples at:
-    // https://github.com/gvolpe/advanced-http4s/blob/master/src/main/scala/com/github/gvolpe/fs2/PubSub.scala
-    // It's not very clear why we need to run this inside a Scheduler
-    Scheduler[IO](corePoolSize = 4).flatMap { _ =>
+        wc <- Resource.liftF(serverConf)
+        ac <- Resource.liftF(authConf)
+        as <- Resource.liftF(authService(ac))
+        _  <- Resource.liftF(logStart.run(wc))
+        _  <- Resource.liftF(logToClients(out))
+        _  <- redirectWebServer(wc)
+        _  <- webServer(as, in, out, et, cr, bec)(wc)
+      } yield ()
+
+    val r: Resource[IO, ExitCode] =
       for {
-        cli    <- Http1Client.stream[IO]()
-        inq    <- Stream.eval(async.boundedQueue[IO, executeEngine.EventType](10))
-        out    <- Stream.eval(async.topic[IO, SeqexecEvent](NullEvent))
-        pe     <- Stream.eval(PrometheusExportService.build[IO])
-        engine <- Stream.eval(engineIO(cli, pe.collectorRegistry))
-        web    <- Stream.eval(webServerIO(inq, out, engine, pe))
-        exit   <- Stream(
-                    engine.eventStream(inq).to(out.publish),
-                    web
-                  ).join(2).drain ++ Stream.emit(ExitCode.Success)
-      } yield exit
-    }
+        cli    <- AsyncHttpClient.resource[IO]()
+        inq    <- Resource.liftF(Queue.bounded[IO, executeEngine.EventType](10))
+        out    <- Resource.liftF(Topic[IO, SeqexecEvent](NullEvent))
+        cr     <- Resource.liftF(IO(new CollectorRegistry))
+        engine <- Resource.liftF(engineIO(cli, cr))
+        bec    <- blockingExecutionContext
+        _      <- webServerIO(inq, out, engine, cr, bec)
+        _      <- Resource.liftF(engine.eventStream(inq).to(out.publish).compile.drain.start)
+      } yield ExitCode.Success
+
+    r.use(_ => IO.never)
```

Comment:
What does this do?

Reply:
The resource is a server running asynchronously on another fiber (logical thread). So here we use `IO.never` to halt the current fiber forever, or until it's cancelled by the user hitting ^C, at which point the cleanup computation runs and shuts everything down.